src/mam4xx/mo_sethet.hpp

```diff
@@ -0,0 +1,181 @@
+#ifndef MAM4XX_MO_SETHET_HPP
+#define MAM4XX_MO_SETHET_HPP
+
+#include <haero/math.hpp>
+#include <mam4xx/gas_chem.hpp>
+#include <mam4xx/mam4_types.hpp>
+#include <mam4xx/mo_chm_diags.hpp>
+#include <mam4xx/utils.hpp>
+
+namespace mam4 {
+
+namespace mo_sethet {
+
+// const int ktop = ConvProc::Config::ktop;
+// int ktop = 47; // BAD_CONSTANT: only true for nlev == 72
+constexpr Real avo = haero::Constants::avogadro;
+const Real pi = haero::Constants::pi;
+constexpr Real rgrav =
+    mo_chm_diags::rgrav; // reciprocal of acceleration of gravity ~ m/s^2
+constexpr const int gas_pcnst = gas_chemistry::gas_pcnst;
+const Real boltz_cgs = haero::Constants::boltzmann * 1.e7; // erg/K
+// number of vertical levels
+constexpr const int pver = mam4::nlev;
+
+// FIXME: BAD CONSTANT
+constexpr const Real mass_h2o = 18.0;   // mass of water vapor [amu]
+constexpr const Real cm3_2_m3 = 1.0e-6; // convert cm^3 to m^3
+constexpr const Real liter_per_gram = 1.0e-3;
+constexpr const Real avo2 =
+    avo * liter_per_gram * cm3_2_m3; // [liter/gm/mol*(m/cm)^3]
+
+using Real = haero::Real;
+using View1D = DeviceType::view_1d<Real>;
+
+//=================================================================================
+KOKKOS_INLINE_FUNCTION
+void calc_het_rates(Real satf, // saturation fraction in cloud //in
+                    Real rain, // rain rate [molecules/cm^3/s] //in
+                    Real xhen, // henry's law constant // in
+                    Real tmp_hetrates, Real work1, Real work2, // in
+                    Real &het_rates) // rainout loss rates [1/s]// out
+{
+  //-----------------------------------------------------------------
+  // calculate het_rates
+  // input arguments are different for different species
+  //-----------------------------------------------------------------
+  Real work3;
+  Real h2o_mol = 1.0e3 / mass_h2o; // [gm/mol water]
+  // FIXME: BAD CONSTANT should haero::Constants::molec_weight_h2o be used
+  // somewhere here instead?
```

Comment:
Perhaps (and maybe even "probably"), but marking this as a `BAD CONSTANT` is sufficient for now.